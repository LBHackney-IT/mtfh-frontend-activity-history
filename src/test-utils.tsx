import { Router, Route } from 'react-router-dom';
import React from 'react';
import fetch from 'node-fetch';
import { rest } from 'msw';
import { createMemoryHistory, MemoryHistory } from 'history';
import { render, RenderResult } from '@testing-library/react';
import { queries } from '@mtfh/common';
import { server } from './mocks';

beforeAll(() => {
    server.listen();
});

afterEach(() => {
    server.resetHandlers();
});

afterAll(() => {
    server.close();
});

interface RouteRenderConfig {
    url: string;
    path: string;
    query: keyof typeof queries;
}

export const routeRender = (
    component: JSX.Element,
    options?: Partial<RouteRenderConfig>
): [RenderResult, MemoryHistory] => {
    const config: RouteRenderConfig = {
        url: '/person/be8c805c-b1de-11eb-8529-0242ac130003',
        path: '/person/:personId',
        query: 'lg',
        ...options,
    };
    const history = createMemoryHistory();
    history.push(config.url);
    return [
        render(
            <Router history={history}>
                <Route path={config.path}>{component}</Route>
            </Router>
        ),
        history,
    ];
};

export const get = (path: string, data: unknown, code = 200): void => {
    server.use(
        rest.get(path, (req, res, ctx) => {
            return res(ctx.status(code), ctx.json(data));
        })
    );
};
