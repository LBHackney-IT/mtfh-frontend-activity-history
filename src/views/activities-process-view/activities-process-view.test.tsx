import React from "react";

import { screen, waitFor } from "@testing-library/react";

import { routeRender } from "../../test-utils";

import { ActivitiesProcessView } from ".";

import { locale } from "@services";

test("renders the activities view", async () => {
  const processName = "soletojoint";
  const [{ container }] = routeRender(<ActivitiesProcessView activityName="process" />, {
    url: `/activities/process/${processName}/123`,
  });
  expect(container).toMatchSnapshot();

  await waitFor(() =>
    expect(screen.getAllByRole("heading")[0]).toHaveTextContent(
      locale.activities.pageTitle,
    ),
  );
});
