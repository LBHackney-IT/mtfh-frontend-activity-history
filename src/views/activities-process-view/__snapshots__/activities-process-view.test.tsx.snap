// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the activities view 1`] = `
<div>
  <div
    data-testid="process-activities"
  >
    <h1
      class="lbh-heading-h1"
    >
      Activity history
    </h1>
    <h2
      class="lbh-heading-h2"
    >
      Sole tenant requests a joint tenure
    </h2>
    <h2
      class="lbh-heading-h2"
    >
      Tenure payment ref
       
      1ifr0k2gtc
    </h2>
    <h2
      class="lbh-heading-h2"
    >
      5130 Wyman Ville, 37849
    </h2>
    <h2
      class="lbh-heading-h2"
    >
      <a
        class="govuk-link lbh-link lbh-link--no-visited-state"
        href="/person/1c6b13f0-24cd-4d1b-a922-87e573d9d42e"
      >
        Global Interactions Architect Desmond Erdman
      </a>
    </h2>
    <div>
      <table
        class="govuk-table lbh-table"
      >
        <thead
          class="govuk-table__head"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Date
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Category
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Case details
            </th>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              By
            </th>
          </tr>
        </thead>
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row govuk-table__row mtfh-activity-history"
          >
            <td
              class="govuk-table__cell"
            >
              <div>
                <p>
                  19/09/19
                </p>
                <p>
                  15:12
                </p>
              </div>
            </td>
            <td
              class="govuk-table__cell"
            >
              Edit to person
            </td>
            <td
              class="govuk-table__cell"
            >
              <div>
                <p>
                  <b>
                    First name
                  </b>
                </p>
                <p>
                  Previously:
                   
                  <b>
                    Susan
                  </b>
                </p>
                <p>
                  Changed to:
                   
                  <b>
                    Susanna
                  </b>
                </p>
              </div>
            </td>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Paul Fox
            </th>
          </tr>
          <tr
            class="govuk-table__row govuk-table__row mtfh-activity-history"
          >
            <td
              class="govuk-table__cell"
            >
              <div>
                <p>
                  19/09/19
                </p>
                <p>
                  15:12
                </p>
              </div>
            </td>
            <td
              class="govuk-table__cell"
            >
              Edit to person
            </td>
            <td
              class="govuk-table__cell"
            >
              <p>
                <b>
                  Person created
                </b>
              </p>
            </td>
            <th
              class="govuk-table__cell govuk-table__header"
            >
              Paul Fox
            </th>
          </tr>
        </tbody>
      </table>
      <nav
        class="lbh-simple-pagination"
      >
        <a
          class="lbh-simple-pagination__link lbh-simple-pagination__link--next"
          rel="next"
        >
          Next
          <svg
            fill="none"
            height="19"
            viewBox="0 0 11 19"
            width="11"
          >
            <path
              d="M1 18L9 9.5L1 1"
              stroke-width="2"
            />
          </svg>
        </a>
      </nav>
    </div>
    <a
      class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
      href="/processes/soletojoint/123"
    >
      Close activity history
    </a>
  </div>
</div>
`;
